contrib/platform/src/com/sun/jna/platform/win32/PdhUtil.java

```diff
@@ -0,0 +1,160 @@
+        if (charToBytes == 1) {
+            return mem.getString(0);
+        } else {
+            return mem.getWideString(0);
+        }
+    }
+
+    /**
+     * Utility method to call Pdh's PdhEnumObjectItems that allocates the
+     * required memory for the mszCounterList and mszInstanceList parameters
+     * based on the type mapping used, calls to PdhEnumObjectItems, and returns
+     * the received lists of strings.
+     * 
+     * @param szDataSource
+     *            String that specifies the name of the log file used to
+     *            enumerate the counter and instance names. If NULL, the
+     *            function uses the computer specified in the szMachineName
+     *            parameter to enumerate the names.
+     * @param szMachineName
+     *            String that specifies the name of the computer that contains
+     *            the counter and instance names that you want to enumerate.
+     *            Include the leading slashes in the computer name, for example,
+     *            \\computername. If the szDataSource parameter is NULL, you can
+     *            set szMachineName to NULL to specify the local computer.
+     * @param szObjectName
+     *            String that specifies the name of the object whose counter and
+     *            instance names you want to enumerate.
+     * @param dwDetailLevel
+     *            Detail level of the performance items to return. All items
+     *            that are of the specified detail level or less will be
+     *            returned.
+     * @return Returns a List with two elements. Index 0 contains a List of
+     *         Strings of the counters for the object. Index 1 contains a List
+     *         of Strings of the instances of the object.
+     */
+    public static List<List<String>> PdhEnumObjectItems(String szDataSource, String szMachineName, String szObjectName,
+            int dwDetailLevel) {
+        int charToBytes = Boolean.getBoolean("w32.ascii") ? 1 : Native.WCHAR_SIZE;
+
+        // Call once to get string lengths
+        DWORDByReference pcchCounterListLength = new DWORDByReference(new DWORD(0));
+        DWORDByReference pcchInstanceListLength = new DWORDByReference(new DWORD(0));
+        Pdh.INSTANCE.PdhEnumObjectItems(szDataSource, szMachineName, szObjectName, null, pcchCounterListLength, null,
+                pcchInstanceListLength, dwDetailLevel, 0);
+
+        // Allocate memory and call again to populate strings
+        Memory mszCounterList = new Memory(pcchCounterListLength.getValue().intValue() * charToBytes);
+        Memory mszInstanceList = new Memory(pcchInstanceListLength.getValue().intValue() * charToBytes);
+        Pdh.INSTANCE.PdhEnumObjectItems(szDataSource, szMachineName, szObjectName, mszCounterList,
+                pcchCounterListLength, mszInstanceList, pcchInstanceListLength, dwDetailLevel, 0);
+
+        // Fetch counters
+        List<String> counters = new LinkedList<>();
```

Comment:
In another issue it was noted, that the LinkedList is not the best list to choose. I suggest to switch to ArrayList.

Reply:
Sure, I'll switch. I'm curious the logic, though?  I thought LinkedList was faster for adding (particularly when the final length is unknown) and ArrayList for getting.